src/board.py

```diff
@@ -24,14 +25,47 @@ class Direction(IntEnum):
 
 def build():
     fences = [[Direction.NO for i in range(FENCE_SIZE)] for j in range(FENCE_SIZE)]
-    for i in range(10):
-        center_x = randint(0, FENCE_SIZE - 1)
-        center_y = randint(0, FENCE_SIZE - 1)
-        direction = randint(Direction.HORIZONTALLY, Direction.VERTICALLY)
-        fences[center_x][center_y] = direction
+    for i in range(FENCE_NUMBER):
+        added = False
+        while not added:
+            center_x = randint(0, FENCE_SIZE - 1)
+            center_y = randint(0, FENCE_SIZE - 1)
```

Comment:
The variables `center_*` are defined with each loop, but they can be defined above.
This is micro optimization, but it can also helps with code readability.